src/test/java/uk/gov/hmcts/reform/cwrdapi/controllers/StaffRefDataControllerTest.java

```diff
@@ -152,6 +217,379 @@ void shouldFetchEmptyUserTypes() {
+        searchResponse.setUserType("1");
+        searchReq = SearchRequest.builder()
+                .userType(userType)
+                .build();
+
+        advancedSearchResponse = new ResponseEntity<>(List.of(searchResponse), null, HttpStatus.OK);
+        when(staffRefDataService.retrieveStaffProfile(eq(searchReq), Mockito.any(PageRequest.class)))
+                .thenReturn(advancedSearchResponse);
+        ResponseEntity<List<SearchStaffUserResponse>> response = staffRefDataController
+                .searchStaffProfile(1,2, searchReq);
+        assertNotNull(response);
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        verify(staffRefDataService, times(1))
+                .retrieveStaffProfile(eq(searchReq), Mockito.any(PageRequest.class));
+
+        assertTrue(validateSearchUserProfileResponse(response,searchReq));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"abc","*__","sdf sdf"})
+    void should_return_staffProfiles_search_with_invalid_input_usertype_status_code_400(String userType) {
+
+        searchReq = SearchRequest.builder()
+                .userType(userType)
+                .build();
+
+        assertThrows(InvalidRequestException.class, () -> staffRefDataController
+                .searchStaffProfile(2,2, searchReq));
+        verify(staffRefDataService, times(0))
+                .retrieveStaffProfile(eq(searchReq), Mockito.any(PageRequest.class));
+    }
+
+
+
+    @ParameterizedTest
+    @ValueSource(strings = {"1"})
+    void should_return_staffProfiles_search_with_jobTitle_status_code_200(String jobTitileId) {
+        List<Role> roles = List.of(Role.builder().roleId("1").roleName("avd").build());
+        searchResponse.setRoles(roles);
+        searchReq = SearchRequest.builder()
+                .jobTitle(jobTitileId)
+                .build();
+
+        advancedSearchResponse = new ResponseEntity<>(List.of(searchResponse), null, HttpStatus.OK);
+        when(staffRefDataService.retrieveStaffProfile(eq(searchReq), Mockito.any(PageRequest.class)))
+                .thenReturn(advancedSearchResponse);
+        ResponseEntity<List<SearchStaffUserResponse>> response =
+                staffRefDataController.searchStaffProfile(1,2, searchReq);
+        assertNotNull(response);
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        verify(staffRefDataService, times(1))
+                .retrieveStaffProfile(eq(searchReq), Mockito.any(PageRequest.class));
+
+        assertTrue(validateSearchUserProfileResponse(response,searchReq));
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {"avdx","_*kd"})
+    void should_return_staffProfiles_search_with_invalid_input_jobTitle_status_code_400(String jobTitileId) {
+        searchReq = SearchRequest.builder()
+                .jobTitle(jobTitileId)
+                .build();
+
+        assertThrows(InvalidRequestException.class, () -> staffRefDataController
```

Comment:
can you check for expected error message

Reply:
changes done